regen/embed.pl

```diff
@@ -394,11 +418,28 @@ sub generate_proto_h {
             die_at_end "$plain_func: Function with '...' arguments must have"
                      . " f or F flag";
         }
+
         if ( @attrs ) {
-            $ret .= "\n";
-            $ret .= join( "\n", map { (" " x 8) . $_ } @attrs );
+            if ( @non_debug_attrs ) {
+                $ret = "#${ind}ifdef DEBUGGING\n\n"
+                     . $ret . "\n"
+                     . join( "\n", map { (" " x 8) . $_ } @attrs ) . ";\n"
+                     . "#${ind}else\n\n"
+                     . $ret . "\n"
+                     . join( "\n", map { (" " x 8) . $_ } @attrs,
+                                                          @non_debug_attrs )
+                     . ";\n"
+                     . "#${ind}endif\n";
+            }
+            else {
```

Comment:
Instead of generating duplicate declarations for each function why not use a macro to only generate the attribute for non-DEBUGGING?

This could be generic like a reverse PERL_DEB(), or it could be more specific like PERL_ATTR_NONNULL_NDEBUG().